### PR TITLE
Connect votes to backend

### PR DESCRIPTION
**Pull request by dinojs**
- Voting system works: userID is currently generated in the backend
- Will be fixed: If the page is refreshed arrows highlighting will not remain although the voteCount will be accurate and not allow the user to vote again  thanks to backend limitations.

### CODE REVIEW

File: src/components/comments/CommentForm.jsx [hunk 1]
```diff
@@ -11,16 +11,35 @@ import { LOCAL_HOST } from "../../consts";
...
     super(props);
-    this.state = { userID: null, userName: '', isNewUser: false };
-    // check if user record exist in localStorage
-    if (localStorage.getItem('user') != null) {
-      let storedUser = JSON.parse(localStorage.getItem('user'))
-      this.state.userID = storedUser.userID
-      this.state.userName = storedUser.userName
-      this.setState(storedUser)
+    this.state = { userID: null, userName: '' };
+    /**
```
Comment (KashSingh1) [inline src/components/comments/CommentForm.jsx:16]: Don't use JavaDoc comments in places like this, it's bad styling. Just stick to //

File: src/components/comments/Comment.jsx [hunk 2]
```diff
@@ -16,14 +15,28 @@ class Comment extends Component {
...
+        .post(`${LOCAL_HOST}/api/v1/users`)
+        .then((res) => {
+          console.log(res.data);
+          // TODO: Retrieve userID and put in state
+        })
+        .catch((err) => {
+          console.log(err);
+        });
+      return;
+>>>>>>> 994fd4317125786b10af18c70d2bc219be131232
```
Comment (KashSingh1) [inline src/components/comments/Comment.jsx]: Pick one and remove the other.

File: src/components/comments/CommentBox.jsx [hunk 3]
```diff
@@ -65,71 +65,67 @@ class CommentBox extends React.Component {
...
     this.getComments();
   }
 
-
-
   /**
    * Sorts feedback by the option the user chose to perform.
    */
-  sortComments(e){
+  sortComments(e) {
```
Comment (KashSingh1) [inline src/components/comments/CommentBox.jsx:77]: Pick a better variable name like "sortChoice" or "option". "e" isn't meaningful here.

**Comment by KashSingh1** [comment]
Clear up merge issues; add in more comments where you thinking they are needed, and make sure code passes CI check

**Comment by bbaayezi** [comment]
Known issues:

- Sorting votes function is still buggy.
- After posting new feedback, one cannot immediately vote his feedback due to the absence of feedback ID (it's a backend problem as well, the `addNewFeedback` endpoint should return the feedback ID to the front end).
- After creating a new user credential and store it in the localStorage in `CommentBox` component, the sub component `CommentForm` cannot get cached user credential from localStorage unless refresh the page. This issue should be related to the asynchronous logic, need to find a way to block the render process of `CommentBox` before it gets a response from the `createNewUser` endpoint.
- Voting feedbacks can sometimes failed. Not sure about the real cause.

Persoanl suggestions on solving the problems in the future: model the operation process of front end users with state machines. Consider a better way to manage states across different front end components.

**Review by bbaayezi — APPROVED** [review]

**Pull request merged**